test/unit/RelayerFactories.t.sol

```diff
@@ -0,0 +1,378 @@
+      address(mockAlgebraFactory),
+      abi.encodeWithSignature('poolByPair(address,address)', _baseToken, _quoteToken),
+      abi.encode(_algebraPool)
+    );
+  }
+
+  function _mockToken0(address _token0) internal {
+    vm.mockCall(address(mockAlgebraPool), abi.encodeWithSignature('token0()'), abi.encode(_token0));
+  }
+
+  function _mockToken1(address _token1) internal {
+    vm.mockCall(address(mockAlgebraPool), abi.encodeWithSignature('token1()'), abi.encode(_token1));
+  }
+
+  function _mockSymbol(string memory _symbol) internal {
+    vm.mockCall(address(mockBaseToken), abi.encodeWithSignature('symbol()'), abi.encode(_symbol));
+    vm.mockCall(address(mockQuoteToken), abi.encodeWithSignature('symbol()'), abi.encode(_symbol));
+    vm.mockCall(address(mockAggregator), abi.encodeWithSignature('description()'), abi.encode(_symbol));
+  }
+
+  function _mockDecimals(uint8 _decimals) internal {
+    vm.mockCall(address(mockBaseToken), abi.encodeWithSignature('decimals()'), abi.encode(_decimals));
+    vm.mockCall(address(mockQuoteToken), abi.encodeWithSignature('decimals()'), abi.encode(_decimals));
+    vm.mockCall(address(mockAggregator), abi.encodeWithSignature('decimals()'), abi.encode(_decimals));
+  }
+}
+
+contract Unit_CamelotRelayerFactory_Constructor is Base {
+  event AddAuthorization(address _account);
+
+  modifier happyPath() {
+    vm.startPrank(user);
+    _;
+  }
+
+  function test_Emit_AddAuthorization() public happyPath {
+    vm.expectEmit();
+    emit AddAuthorization(user);
+
+    camelotRelayerFactory = new CamelotRelayerFactory();
+  }
+}
+
+contract Unit_RelayerFactory_DeployCamelotRelayer is Base {
+  event NewAlgebraRelayer(address indexed _relayer, address _baseToken, address _quoteToken, uint32 _quotePeriod);
+
+  modifier happyPath(string memory _symbol, uint8 _decimals) {
+    vm.startPrank(authorizedAccount);
+    _assumeHappyPath(_decimals);
+    _mockValues(_symbol, _decimals);
+    _;
+  }
+
+  function _assumeHappyPath(uint8 _decimals) internal pure {
+    vm.assume(_decimals <= 18);
+  }
+
+  function _mockValues(string memory _symbol, uint8 _decimals) internal {
+    _mockGetPool(address(mockBaseToken), address(mockQuoteToken), address(mockAlgebraPool));
+    _mockToken0(address(mockBaseToken));
+    _mockToken1(address(mockQuoteToken));
+    _mockSymbol(_symbol);
+    _mockDecimals(_decimals);
+  }
+
+  function test_Revert_Unauthorized(uint32 _quotePeriod) public {
+    vm.expectRevert('Unauthorized');
+
+    camelotRelayerFactory.deployAlgebraRelayer(
+      SEPOLIA_ALGEBRA_FACTORY, address(mockBaseToken), address(mockQuoteToken), _quotePeriod
+    );
+  }
+
+  function test_Deploy_RelayerChild(
+    uint32 _quotePeriod,
+    string memory _symbol,
+    uint8 _decimals
+  ) public happyPath(_symbol, _decimals) {
+    vm.expectEmit();
+    emit NewAlgebraRelayer(
+      address(0x7F85e9e000597158AED9320B5A5E11AB8cC7329A), address(mockBaseToken), address(mockQuoteToken), _quotePeriod
```

Comment:
What address is this? Can you use a variable here instead?

Reply:
this is the address that the new contract will be deployed to.   I just used it as shorthand for computing the address ahead of time or using create2.  I can hard code it into a global variable to make it cleaner.